calico/felix/test/test_felix.py

```diff
@@ -39,8 +37,19 @@
 class TestException(Exception):
     pass
 
-
 class TestBasic(BaseTestCase):
+    def setUp(self):
+        super(TestBasic, self).setUp()
+        global real_etcd
+        real_etcd = sys.modules.get('etcd', None)
+        sys.modules['etcd'] = stub_etcd
+
+    def tearDown(self):
+        super(TestBasic, self).tearDown()
+        if real_etcd is None:
+            sys.modules.pop('etcd')
+        else:
+            sys.modules['etcd'] = real_etcd
```

Comment:
There's no reason to use a `global` here, you can just store it on the test class itself (`self._real_etcd`).